Library/Homebrew/system_command.rb

```diff
@@ -276,9 +273,56 @@ def each_output_line(&block)
   rescue Interrupt
     Process.kill("INT", raw_wait_thr.pid) if raw_wait_thr && !sudo?
     raise Interrupt
-  rescue SystemCallError => e
-    @status = $CHILD_STATUS
-    @output << [:stderr, e.message]
+  ensure
+    raw_stdin&.close
+    raw_stdout&.close
+    raw_stderr&.close
+  end
+
+  sig {
+    params(
+      env:        T::Hash[String, String],
+      executable: String,
```

Comment:
I think this could also be a `Pathname`.

Reply:
It gets converted here: https://github.com/Homebrew/brew/blob/ce31ab8b4d8ada2592587180f18b4855f5ef5139/Library/Homebrew/system_command.rb#L148

and retrieved here: 
https://github.com/Homebrew/brew/blob/ce31ab8b4d8ada2592587180f18b4855f5ef5139/Library/Homebrew/system_command.rb#L233